components/datetime/src/pattern/names.rs

```diff
@@ -2826,6 +2826,200 @@ where
+    ///     None
+    /// );
+    /// ```
+    pub fn get_month(&self, month_code: MonthCode, length: MonthNameLength) -> Option<&str> {
+        let borrowed = self.inner.as_borrowed();
+        let month_names = borrowed.month_names.get_with_variables(length)?;
+
+        let (month_number, is_leap) = month_code.parsed()?;
+        let month_index = usize::from(month_number.checked_sub(1)?);
+
+        match month_names {
+            MonthNames::Linear(linear) => {
+                if is_leap {
+                    None
+                } else {
+                    linear.get(month_index)
+                }
+            }
+            MonthNames::LeapLinear(leap_linear) => {
+                let num_months = leap_linear.len() / 2;
+                if is_leap {
+                    leap_linear.get(month_index + num_months)
+                } else if month_index < num_months {
+                    leap_linear.get(month_index)
+                } else {
+                    None
+                }
+            }
+            MonthNames::LeapNumeric(_) => None,
+        }
+    }
+}
+
+impl<C: CldrCalendar, FSet: DateTimeNamesMarker> FixedCalendarDateTimeNames<C, FSet>
+where
+    FSet::YearNames: NamesContainer<
+        YearNamesV1,
+        YearNameLength,
+        Container = DataPayloadWithVariables<YearNamesV1, YearNameLength>,
+    >,
+{
+    /// Gets the era name for the specified length and era if the data is loaded.
+    ///
+    /// Returns `Some` if the data for the specified length is loaded and the era is valid,
+    /// or `None` if not loaded or the era is invalid.
+    ///
+    /// Note: This is a low-level API. The `era_year` parameter is typically obtained from
+    /// calendar date operations. Most users should use the formatting APIs instead.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::calendar::{Date, Gregorian};
+    /// use icu::datetime::pattern::{FixedCalendarDateTimeNames, YearNameLength};
+    /// use icu::locale::locale;
+    ///
+    /// let mut names =
+    ///     FixedCalendarDateTimeNames::<Gregorian>::try_new(locale!("en").into())
+    ///         .unwrap();
+    ///
+    /// // Get era from a date
+    /// let date = Date::try_new_gregorian(2024, 1, 1).unwrap();
+    /// let era_year = date.era_year();
+    ///
+    /// // Before loading data, the getter returns None:
+    /// assert_eq!(names.get_era(&era_year, YearNameLength::Wide), None);
+    ///
+    /// // Load the year names:
+    /// names.include_year_names(YearNameLength::Wide).unwrap();
+    ///
+    /// // Now we can get the era name for the loaded length:
+    /// assert_eq!(names.get_era(&era_year, YearNameLength::Wide), Some("Anno Domini"));
+    ///
+    /// // But other lengths are not loaded:
+    /// assert_eq!(
+    ///     names.get_era(&era_year, YearNameLength::Abbreviated),
+    ///     None
+    /// );
+    /// ```
+    pub fn get_era(&self, era_year: &EraYear, length: YearNameLength) -> Option<&str> {
```

Comment:
issue: clients won't know how to construct an `EraYear` correctly, so this method is only useful for eras returned by a `Date`.

do we want to add support for accept _era codes_ (like `ce`) here? we would need some kind of integration with `icu::calendar`.